### PR TITLE
check it MessagePort and Worker are defined in getTargetInfoFromDef

### PR DESCRIPTION
**Pull request by artem-karnaukh-wix**
should solve - https://github.com/wix/pm-rpc/issues/31

### CODE REVIEW

File: src/pm-rpc/rpc.js [hunk 1]
```diff
@@ -9,9 +9,9 @@ import {serialize as serializeError} from './privates/errorSerializer'
 
 const getTargetInfoFromDef = ({target, initiator}) => {
   switch (true) {
-    case target instanceof MessagePort:
+    case typeof MessagePort !== 'undefined' && target instanceof MessagePort:
       return {target}
-    case target instanceof Worker:
+    case typeof Worker !== 'undefined' && target instanceof Worker:
```
Comment (maxaon) [inline src/pm-rpc/rpc.js:14]: Please add comment with description why it's required

**Review by maxaon — CHANGES_REQUESTED** [review]
Please add record to changelog

File: src/pm-rpc/rpc.js (hunk 1 above)
Comment (artem-karnaukh-wix) [reply src/pm-rpc/rpc.js:16]: done

File: src/pm-rpc/rpc.js [hunk 2]
```diff
@@ -9,9 +9,11 @@ import {serialize as serializeError} from './privates/errorSerializer'
 
 const getTargetInfoFromDef = ({target, initiator}) => {
   switch (true) {
-    case target instanceof MessagePort:
+    // MessagePort is undefined in Safari in the code that is executed inside a web worker
```
Comment (maxaon) [inline src/pm-rpc/rpc.js:12]:
```suggestion
    // In case of MessagePort is undefined
```

**Review by maxaon — APPROVED** [review]

**Review by rbh3 — APPROVED** [review]
LGTM

**Review by sdoomz — COMMENTED** [review]
As I understand, it suppose to fix https://github.com/wix/pm-rpc/issues/31, @artem-karnaukh-wix could you add it to the description?

**Comment by sdoomz** [comment]
@vadimavdeev also take a look please

**Review by sdoomz — APPROVED** [review]

File: src/pm-rpc/rpc.js [hunk 3]
```diff
@@ -9,9 +9,11 @@ import {serialize as serializeError} from './privates/errorSerializer'
 
 const getTargetInfoFromDef = ({target, initiator}) => {
   switch (true) {
-    case target instanceof MessagePort:
+    // In case of MessagePort is undefined
+    case typeof MessagePort !== 'undefined' && target instanceof MessagePort:
```
[the remark this answers is not in the document]
Comment (maxaon) [reply src/pm-rpc/rpc.js:13]: We added it for safeness. But if it's not defined probably whole pmrpc will not work.

**Review by vadimavdeev — APPROVED** [review]

**Pull request merged**